core-plugins/src/main/java/co/cask/hydrator/plugin/common/FileUtililty.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.hydrator.plugin.common;
+
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+import java.io.IOException;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+/**
+ * FileSystem Utility class doing various operations on file.
+ */
+public final class FileUtililty {
+
+  private FileUtililty() {
+    throw new AssertionError("Should not instantiate static utility class.");
+  }
+
+  /**
+   * Method to archive source file into target
+   * @param fileSystem - file system
+   * @param sourceFile - source file to be archived
+   * @param targetFile - target file where source file to be archived.
+   * @param deleteSource - True if file to be deleted after archiving otherwise false.
+   * @throws IOException - IOException occurred while archiving file.
+   */
+  public static void archiveFile(FileSystem fileSystem, Path sourceFile, Path targetFile, boolean deleteSource) throws
```

Comment:
why is this class needed? This method is only used once, right? Unless we find more usages of this method, we shouldn't create static utility classes for it. Move it to XMLRecordReader?